report_custom_filename/controllers/reports.py

```diff
@@ -17,10 +17,10 @@ def index(self, action, token):
         context = dict(http.request.context)
         context.update(action["context"])
         report_xml = http.request.session.model('ir.actions.report.xml')
```

Comment:
Thank you Pedro! Didn't know that. Changed it and with another minor change and it works fine. Another PR on the way...